### PR TITLE
Collect CRD descriptions from cluster

### PR DESCRIPTION
**Pull request by danbudris**
https://github.com/aws/eks-anywhere/issues/349

This PR will allow us to collect detailed information about CRDs and include the JSON descriptions in the support-bundle archive. This is critical for troubleshooting EKS-A clusters, and a pre-requisite for adding detailed analyzers for cluster state.

This is done by leveraging the `support-bundle` `run` collectors to run a pod on the cluster and invoking `kubectl get MYCRD.dan.dev -o json --all-namespaces` on the CRDs in question.

Specifically, this PR includes collectors for a first set of 'default' CRDs:
- `clusters.anywhere.eks.amazonaws.com`,
- `bundles.anywhere.eks.amazonaws.com`,
- `clusters.cluster.x-k8s.io`,
- `machinedeployments.cluster.x-k8s.io`,
- `machines.cluster.x-k8s.io`,

Additionally, it introduces an `eksa-diagnostics` namespace and associated RBAC so that we can execute pod-based collectors (`copyFromHost`, `run` collectors) in a controlled environment. The `run` collectors use the default service account in a given namespace, and can't be configured otherwise, so using our own ephemeral namespace/rbac will allow us to provide exactly the permissions we need to the `run` collector pods without binding roles to existing default service accounts in, say, `eksa-system`.

This new ephemeral `eksa-diagnostics` namespace is created when the command is invoked and deleted when we clean up. This uses two new Delete methods in the `kubectl` executable. Creation/delete are retried, but we don't exit on error -- this allows us to run `support-bundle` on a flaky cluster that, say, can't create new objects via the API, for example, and so can't create the namespace; collectors for `support-bundle` are best-effort, and even if a collector fails (for example, collecting host logs because you can't launch a new pod) the other collectors will continue and execute (like listing CRDs via the API and ensuring they're present, or collecting logs from existing pods).

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/365#" title="Author self-approved">danbudris</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [danbudris]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/diagnostics/collectors.go
Comment (danbudris) [inline pkg/diagnostics/collectors.go:42]: this whole block is mostly just stuff being moved around

File: pkg/diagnostics/collectors.go
Comment (g-gaston) [inline pkg/diagnostics/collectors.go:208]: curious why in all namespaces? Don't we know already what are the namespaces where we will find each CRD?

I personally don't think it's a big deal, buy that would allow us to scope down the permissions for the service account to get and list only on specific namespaces

**Review by g-gaston — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/365#" title="Author self-approved">danbudris</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/365#pullrequestreview-773010754" title="Approved">g-gaston</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [danbudris,g-gaston]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/diagnostics/collectors.go
Comment (danbudris) [reply pkg/diagnostics/collectors.go:208]: yeah I thought the increased visibility helped to ensure we wouldn't miss something 'out-of-place' by only focusing on the intended namespaces; for example if somehow one of our resources ended up in the wrong namespace, or a customer added a conflicting resource in another namespace, it'd be interesting to see it

File: pkg/diagnostics/runsupportbundle.go
Comment (danbudris) [reply pkg/diagnostics/runsupportbundle.go:239]: ya sure

File: pkg/diagnostics/collectors.go
Comment (danbudris) [reply pkg/diagnostics/collectors.go:193]: yeah in kubectl we construct the resource name strings using the Group Version/Group from the appropriate API package. We could do that here, export the vars form `kubectl`, or include `var`s in the constants package

File: pkg/diagnostics/collectors.go
Comment (danbudris) [reply pkg/diagnostics/collectors.go:27]: can take this on in the next pr....it gets a little messy with the current structure, but I'll be int here adding collectors

File: pkg/diagnostics/collectors.go
Comment (danbudris) [reply pkg/diagnostics/collectors.go:193]: I can include this in a folow-on PR. I want to think about where these should live

**Pull request merged**
